tests/controllers/QueryContrtollerTest.php

```diff
@@ -0,0 +1,291 @@
+<?php
+
+namespace Tests\Controllers;
+
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+
+class QueryContrtollerTest extends \TestCase
+{
+    use DatabaseTransactions;
+
+    /**
+     * [$service description]
+     * @var \App\Models\Service
+     */
+    protected $service;
+    /**
+     * store the optional channel labels
+     * @var array
+     */
+    protected $channelKeys = [];
+
+    /**
+     * store the optional format key
+     * allowed: [null, 'html', 'json-ls', 'json']
+     * @var string
+     */
+    protected $format = null;
+
+    /**
+     * setup for each test
+     */
+    public function setup()
+    {
+        parent::setUp();
+        $this->service = \App\Models\Service::first();
+
+        $this->channelKeys = $this->service->channels()->pluck('label')->all();
+    }
+
+    public function requestTypeProvider()
+    {
+        $dateParam = date('d-m-Y', strtotime('tomorrow'));
+        return [
+            [['q' => 'now']],
+            [['q' => 'day', 'date' => $dateParam]],
+            [['q' => 'week']],
+            [['q' => 'fullWeek', 'date' => $dateParam]],
+        ];
+    }
+
+    /**
+     * @test
```

Comment:
Start methods that perform tests always with `test`:
* Clear what the test methods are.
* No need to add `@test` annotation.

```php
public function testValidateNoServiceUriParameterIsAFail()
```